lib/tessel/wifi.js

```diff
@@ -71,58 +70,56 @@ function compareBySignal(a, b) {
+  var status = 'Wifi connection successful.';
 
-  return new Promise(function(resolve, reject) {
-    if (!ssid || !password) {
-      return reject(new Error('Invalid credentials. Must set ssid and password'));
+  if (password && !security) {
+    security = 'psk2';
+  }
+
+  if (password && security) {
+    status += ' SSID: ' + ssid + ', password ' + password + ', security mode: ' + security;
+  } else if (!password && (!security || security === 'none')) {
+    security = 'none';
+    status += ' SSID: ' + ssid;
+  }
+
+  var setSSID = () => this.simpleExec(commands.setNetworkSSID(ssid));
+
+  var setNetworkPassword = () => this.simpleExec(commands.setNetworkPassword(password));
+
+  var setNetworkSecurity = () => this.connection.exec(commands.setNetworkEncryption(security));
+
+  var turnWifiOn = () => this.setWiFiState(true);
+
+  var logStatus = () => logs.info(status);
+
+  var setup = () => {
+    if (password) {
+      return setSSID()
+        .then(setNetworkPassword)
+        .then(setNetworkSecurity);
+    } else {
+      return setSSID()
+        .then(setNetworkSecurity);
     }
+  };
 
-    logs.info('Setting SSID:', ssid, 'and password:', password);
-
-    // Set the network SSID
-    return self.simpleExec(commands.setNetworkSSID(ssid))
-      .then(function() {
-        // Then set the password
-        return self.simpleExec(commands.setNetworkPassword(password));
-      })
-      .then(function() {
-        // Then set the encryption
-        return self.connection.exec(commands.setNetworkEncryption('psk2'));
-      })
-      .then(function() {
-        // Then make sure wireless is enabled
-        return self.setWiFiState(true);
-      })
-      .then(function() {
-        // Then set the new credientials
-        logs.info('Credentials set!');
-        resolve();
-      });
-  });
+  return setup()
+    .then(turnWifiOn)
+    .then(logStatus);
 };
 
 Tessel.prototype.resetMDNS = function() {
-  var self = this;
-
-  return self.simpleExec(commands.callMDNSDaemon('restart'))
-    .then(function restarted() {
-      return self.simpleExec(commands.callTesselMDNS('restart'));
-    });
+  return this.simpleExec(commands.callMDNSDaemon('restart'))
+    .then(() => this.simpleExec(commands.callTesselMDNS('restart')));
 };
 
 Tessel.prototype.setWiFiState = function(enable) {
-  var self = this;
-  return self.simpleExec(commands.turnOnWifi(enable))
-    .then(function stateSet() {
-      return self.simpleExec(commands.commitWirelessCredentials());
-    })
-    .then(function committed() {
-      return self.simpleExec(commands.reconnectWifi());
-    })
-    .then(function log() {
-      logs.info('Wifi', enable ? 'Enabled.' : 'Disabled.');
-    });
+  return this.simpleExec(commands.turnOnWifi(enable))
+    .then(() => this.simpleExec(commands.commitWirelessCredentials()))
+    .then(() => this.simpleExec(commands.reconnectWifi()))
+    .then(() => logs.info('Wifi', enable ? 'Enabled.' : 'Disabled.'));
```

Comment:
Just to share, @reconbot @brianloveswords and I just sat together and admired how nice the changes here are and how much easier they are to follow and read—great work!